src/server/analytic_services/prometheus_reporting.js

```diff
@@ -2,359 +2,423 @@
-}, {
-    metric_type: 'Gauge',
-    metric_variable: 'rebuild_time',
-    configuration: {
-        name: get_metric_name('rebuild_time'),
-        help: 'Rebuild Time',
-    },
-    generate_default_set: true,
-}, {
-    metric_type: 'Gauge',
-    metric_variable: 'bucket_healthy',
-    configuration: {
-        name: get_metric_name('bucket_status'),
-        help: 'Bucket Health',
-        labelNames: ['bucket_name'],
-    },
-}, {
-    metric_type: 'Gauge',
-    metric_variable: 'bucket_capacity_precent',
-    configuration: {
-        name: get_metric_name('bucket_capacity'),
-        help: 'Bucket Capacity Precent',
-        labelNames: ['bucket_name'],
-    },
-}, {
-    metric_type: 'Gauge',
-    metric_variable: 'bucket_quota_precent',
-    configuration: {
-        name: get_metric_name('bucket_quota'),
-        help: 'Bucket Quota Precent',
-        labelNames: ['bucket_name'],
-    },
-}, {
-    metric_type: 'Gauge',
-    metric_variable: 'resource_healthy',
-    configuration: {
-        name: get_metric_name('resource_status'),
-        help: 'Resource Health',
-        labelNames: ['resource_name'],
-    },
-}, {
-    metric_type: 'Gauge',
-    metric_variable: 'system_links',
-    configuration: {
-        name: get_metric_name('system_links'),
-        help: 'System Links',
-        labelNames: ['resources', 'buckets', 'dashboard'],
-    },
-}, {
-    metric_type: 'Gauge',
-    metric_variable: 'health_status',
-    configuration: {
-        name: get_metric_name('health_status'),
-        help: 'Health status',
-    },
-    generate_default_set: true,
-}]);
-
-
-class PrometheusReporting {
-    static instance() {
-        PrometheusReporting._instance = PrometheusReporting._instance || new PrometheusReporting();
-        return PrometheusReporting._instance;
+
+    export_metrics() {
+        return this._registry.metrics();
     }
+}
 
+// -----------------------------------------
+// Report containing node js metrics
+// collected and provided by the prom client
+// -----------------------------------------
+class NodeJsReport extends BasePrometheusReport {
     constructor() {
-        METRIC_RECORDS.filter(metric => metric.generate_default_set).forEach(metric => {
-            this[`set_${metric.metric_variable}`] = function(data) {
-                if (!this.enabled()) return;
-                this._metrics[metric.metric_variable].set(data);
-            };
+        super();
+
+        this._prom_client.collectDefaultMetrics({
+            register: this._registry,
+            prefix: this.metric_prefix
         });
-        this._prom_client = prom_client;
-        if (this.enabled()) {
-            this.define_metrics();
-        }
     }
 
-    client() {
-        return this._prom_client;
+    get metric_prefix() {
+        return `${super.metric_prefix}${dbg.get_process_name()}_`;
     }
+}
 
-    enabled() {
-        return config.PROMETHEUS_ENABLED;
-    }
+register_report(
+    config.NODEJS_METRICS_REPORT,
```

Comment:
Why do we need to register the base ? its always there if I remember correctly

Reply:
we are not registering the base. We are registering the nodejs report which enables the default metrics.
They are not inheriting from each other

Reply:
I will rephrase, isn't registring a customer collection (so core for example) bring with it the base ?

Reply:
No, not after the refactor. It is a diff report on a diff client registry

Reply:
from the prom client module, I might be mistaken, but I remember adding anything automatically adds the node.js counters

Reply:
Only if you add to the default registry and collect also the nodejs metrics into the same registry (which is the default behavior)